src/components/learning-guide/practice-by-type-button.cjsx

```diff
@@ -16,13 +15,25 @@ module.exports = React.createClass
 
   onClick: ->
     {courseId} = @props
-    all = LearningGuide.Student.store.getSortedSections(@props.courseId)
-    sections = if @props.practiceType is 'weaker' then _.first(all, 4) else _.last(all, 4)
-    page_ids = _.chain(sections).pluck('page_ids').flatten().uniq().value()
-    @context.router.transitionTo('viewPractice', {courseId}, {page_ids})
+    page_ids = LearningGuide.Student.store.getPracticePages(@props.courseId, @props.practiceType)
+    unless _.isEmpty(page_ids)
```

Comment:
nevermind. I need to read a couple of lines below. maybe this could `throw new Error('BUG: ....')`?